src/overlays/actors/ovl_En_Pr2/z_en_pr2.c

```diff
@@ -71,48 +92,677 @@ const ActorInit En_Pr2_InitVars = {
+        } else {
+            func_80A74888(this);
+        }
+    } else {
+        this->unk_204 = 0.02f;
+        func_80A751B4(this);
+    }
+
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 20.0f, 20.0f, 0x1D);
+
+    if (!(this->actor.bgCheckFlags & 0x60)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnPr2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPr2* this = THIS;
+
+    if (this->unk_1E0 < 10) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+
+    if (this->actor.parent != NULL) {
+        Actor* parent = this->actor.parent;
+
+        if ((parent->update != NULL) && (((TempStruct*)parent)->unk_14E > 0)) {
+            ((TempStruct*)parent)->unk_14E--;
+        }
+    }
+}
+
+s32 func_80A7429C(EnPr2* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s16 sp1A;
+    s16 sp18;
+
+    Actor_GetScreenPos(globalCtx, &this->actor, &sp1A, &sp18);
+
+    if ((fabsf(player->actor.world.pos.y - this->actor.world.pos.y) > 160.0f) ||
+        (this->actor.projectedPos.z < -40.0f) || (sp1A < 0) || (sp1A > 320) || (sp18 < 0) || (sp18 > 240)) {
+        return false;
+    }
+
+    if (!(player->stateFlags1 & 0x8000000)) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void func_80A7436C(EnPr2* this, s16 arg1) {
+    s16 sp2E = arg1 - this->actor.world.rot.y;
+
+    if (sp2E > 10000) {
+        sp2E = 10000;
+    } else if (sp2E < -10000) {
+        sp2E = -10000;
+    }
+
+    Math_ApproachF(&this->actor.world.pos.y, this->unk_21C.y, 0.3f, 5.0f);
+
+    if (fabsf(this->actor.world.pos.y - this->unk_21C.y) > 10.0f) {
+        Math_SmoothStepToS(&this->actor.world.rot.x, Math_Vec3f_Pitch(&this->actor.world.pos, &this->unk_21C) * 0.3f,
+                           20, 5000, 500);
+    } else {
+        Math_SmoothStepToS(&this->actor.world.rot.x, 0, 20, 5000, 500);
+    }
+
+    if (fabsf(this->actor.world.rot.y - arg1) < 30.0f) {
+        Math_ApproachZeroF(&this->unk_1FC, 0.5f, 20.0f);
+    } else {
+        Math_ApproachF(&this->unk_1FC, sp2E, 0.5f, 3000.0f);
+    }
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, arg1, 1, 2000, 300);
+}
+
+void func_80A74510(EnPr2* this, s32 arg0) {
+    static AnimationHeader* D_80A75C2C[] = {
+        &object_pr_Anim_004340,
+        &object_pr_Anim_004274,
+        &object_pr_Anim_003904,
+    };
+    static u8 D_80A75C38[] = {
+        0,
+        0,
+        2,
+        0,
+    };
+    f32 sp34 = 1.0f;
+
+    this->unk_210 = arg0;
+    this->unk_1F8 = Animation_GetLastFrame(D_80A75C2C[arg0]);
+    if (this->unk_210 == 3) {
+        sp34 = 0.0f;
+    }
+    Animation_Change(&this->skelAnime, D_80A75C2C[arg0], sp34, 0.0f, this->unk_1F8, D_80A75C38[arg0], 0.0f);
+}
+
+void func_80A745C4(EnPr2* this) {
+    func_80A74510(this, 0);
+    this->unk_1D4 = 0;
+    this->actionFunc = func_80A745FC;
+}
+
+void func_80A745FC(EnPr2* this, GlobalContext* globalCtx) {
+    f32 x, y, z;
+    f32 sqrt;
```

Comment:
```suggestion
    f32 sqrtXYZ;
```
I would at least do something like this